infrastructure/src/main/java/org/corfudb/infrastructure/configuration/ServerConfiguration.java

```diff
@@ -0,0 +1,507 @@
+package org.corfudb.infrastructure.configuration;
+
+
+import io.netty.channel.EventLoopGroup;
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.configuration.ConfigurationException;
+import org.apache.commons.configuration.PropertiesConfiguration;
+import org.corfudb.comm.ChannelImplementation;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+
+/**
+ * This class holds various server configuration parameters.
+ *
+ * <p>Created by maithem on 12/4/19.
+ */
+
+public class ServerConfiguration extends PropertiesConfiguration {
+    // Server general parameters
+    private static final String SERVER_DIR = "serverDirectory";
+    private static final String SINGLE_MODE = "singleMode";
+    private static final String CLUSTER_ID = "clusterId";
+    private static final String NUM_IO_THREADS = "numIOThreads";
+    private static final String HOST_ADDRESS = "hostAddress";
+    private static final String SERVER_PORT = "serverPort";
+    private static final String NETWORK_INTERFACE = "networkInterface";
+    private static final String HANDSHAKE_TIMEOUT = "handshakeTimeout";
+    private static final String METADATA_RETENTION = "metadataRetention";
+    private static final String LOG_LEVEL = "logLevel";
+    private static final String NUM_BASE_SERVER_THREADS = "numBaseServerThreads";
```

Comment:
I think this file should contain the thread prefixes as well. This way we have it all in one place. Easier for people looking at the logs.